controllers/nodehealthcheck_controller.go

```diff
@@ -318,47 +317,88 @@ func (r *NodeHealthCheckReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	// Delete orphaned CRs: they have no node, and Succeeded and NodeNameChangeExpected conditions set to True.
 	// This happens e.g. on cloud providers with Machine Deletion remediation: the broken node will be deleted and
 	// a new node created, with a new name, and no relationship to the old node
-	if err = r.deleteOrphanedRemediationCRs(nhc, append(healthyNodes, unhealthyNodes...), resourceManager, log); err != nil {
+	if err = r.deleteOrphanedRemediationCRs(nhc, append(notMatchingNodes, matchingNodes...), resourceManager, log); err != nil {
 		return result, err
 	}
 
 	// delete remediation CRs for healthy nodes
-	for _, node := range healthyNodes {
+	healthyCount := 0
+	for _, node := range notMatchingNodes {
 		log.Info("handling healthy node", "node", node.GetName())
 		remediationCRs, err := resourceManager.ListRemediationCRs(nhc, func(cr unstructured.Unstructured) bool {
-			return cr.GetName() == node.GetName()
+			return cr.GetName() == node.GetName() && resources.IsOwner(&cr, nhc)
```

Comment:
Isn't this additional check, `resources.IsOwner(&cr, nhc)`, should be done also in the next call of [`ListRemedationCRs`](https://github.com/medik8s/node-healthcheck-operator/pull/238/files#diff-42b14dc2a70877933db4647434fcbe9b65a8859c7ec1911b0c9da7361c8e98b1R413)